### PR TITLE
Jh/mobile pwa

### PR DESCRIPTION
**Pull request by Hiro-Nakamura**
## Release Notes

- Mobile PWA builder

### CODE REVIEW

**Review by guyyoo — APPROVED** [review]
LGTM!

File: api/controllers/mobile/manifest.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// const authLogger = require("../../lib/authLogger.js");
+
+// var inputParams = {
+//    tenant: { string: true, optional: true },
+// };
+
+module.exports = function (req, res) {
+   req.ab.log("mobile/manifest():");
+
+   let tenantID = req.ab.param("tenantID");
```
Comment (nh758) [inline api/controllers/mobile/manifest.js:20]: Why not use the tenant id from the `authTenant` policy? If we're not using maybe we should skip `authtTenant` for this route?

File: api/controllers/mobile/app.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+//    tenant: { string: true, optional: true },
+// };
+
+module.exports = function (req, res) {
+   req.ab.log("mobile/app():");
+
+   var user = req.ab.user;
+   let appID = req.ab.param("ID");
+   let tenantID = req.ab.tenantID;
+   if (!tenantID || tenantID == "??") {
```
Comment (nh758) [inline api/controllers/mobile/app.js:23]: With the changes in #118 we should never get here. Would need to change `authTenant` if we're expecting use a param to resolve the tenant here.

**Review by nh758 — CHANGES_REQUESTED** [review]

File: api/controllers/mobile/version.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// var inputParams = {
+//    tenant: { string: true, optional: true },
+// };
+
+module.exports = function (req, res) {
+   req.ab.log("mobile/version():");
+
+   var user = req.ab.user;
+   if (user) {
+      res.ab.success({ version: "1.0.0" });
```
[the remark this answers is not in the document]
Comment (Hiro-Nakamura) [reply api/controllers/mobile/version.js:23]: oops, totally forgot about that.

File: api/controllers/mobile/manifest.js (hunk 1 above)
Comment (Hiro-Nakamura) [reply api/controllers/mobile/manifest.js:20]: mostly 'cause the route required the `:tenantID` :  `{get} /mobile/app/:tenantID/:ID/manifest.json`

But it does make more sense to pull the authenticated ID from the `authTenant` policy.

**Review by nh758 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**